torch/testing/_internal/common_device_type.py

```diff
@@ -486,6 +486,9 @@ def instantiate_device_type_tests(generic_test_class, scope, except_for=None, on
             continue
         if only_for is not None and base.device_type not in only_for:
             continue
+        # Special-case for ROCm testing -- only test for 'cuda' i.e. ROCm device by default
+        if TEST_WITH_ROCM and except_for is None and only_for is None and base.device_type != 'cuda':
```

Comment:
Is it necessary for `except_for` and `only_for` to be `None`? Should we not include cases where `only_for is not None and base.device_type != 'cuda'` OR where `except_for is not None and base.device != 'cuda'`. In other words, the condition could simply be `if TEST_WITH_ROCM and base.device_type != 'cuda':`?

Reply:
I have updated the condition.  Because the `only_for` and `except_for` conditions are already processed by the time it reaches this check, the only thing left to do for `TEST_WITH_ROCM` is to `continue` if the device is not `'cuda'`.